src/librbd/CMakeLists.txt

```diff
@@ -24,11 +24,13 @@ set(librbd_internal_srcs
   LibrbdWriteback.cc
   MirroringWatcher.cc 
   ObjectMap.cc
-  ObjectWatcher.cc 
   Operations.cc
   Utils.cc
   cache/ImageWriteback.cc
   cache/PassthroughImageCache.cc
+  watcher/Watcher.cc
```

Comment:
Minor: this should be in the "librbd" namespace instead of "librbd::watcher::Watcher". Also, we shouldn't have both a "librbd::watcher" and "librbd::object_watcher" namespace, so consolidate the librbd::object_watcher::Notifier into this new namespace.